lib/ui/views/home/home_view.dart

```diff
@@ -23,34 +26,54 @@ class HomeView extends StatelessWidget {
             color: TextColorDark,
           ),
           actions: [
-            IconButton(
+            RotatedWidget(
+              widgetToAnimate: IconButton(
                 icon: Icon(
                   Icons.cached_rounded,
                   color: TextColorDark,
                 ),
-                onPressed: () => model.doSomething())
+                onPressed: () {
+                  model.getStoriesAndCache();
+                  _animationSyncButtonController?.rotate();
+                },
+              ),
+              animationCallback: (animationCallback) {
+                _animationSyncButtonController = animationCallback;
+              },
+            )
           ],
         ),
         body: model.stories.length == 0
             ? model.loader
             : Padding(
                 padding: const EdgeInsets.all(8.0),
-                child: ListView.builder(
-                  itemCount: model.stories.length,
-                  itemBuilder: (_, index) {
-                    return ListTile(
-                      onTap: () {
-                        model.openStory(model.stories[index].url);
-                      },
-                      title: Text(model.stories[index].title,
-                          style: h4.copyWith(color: TextColorDark)),
-                      subtitle: Text(model.stories[index].by),
-                    );
+                child: NotificationListener<ScrollNotification>(
+                  onNotification: (ScrollNotification scrollInfo) {
+                    if (scrollInfo.metrics.pixels ==
+                        scrollInfo.metrics.maxScrollExtent) {
+                      model.getNext();
```

Comment:
add a `loading more stories` text at the end of the list.

Reply:
bro this for pagination right we can keep it since it gets data from cache